inc/parser/handler.php

```diff
@@ -926,21 +974,22 @@ public function rss($match, $state, $pos) {
      * @param int $pos byte position in the original source file
      * @return bool mode handled?
      */
-    public function externallink($match, $state, $pos) {
-        $url   = $match;
+    public function externallink($match, $state, $pos)
+    {
+        $url = $match;
         $title = null;
 
         // add protocol on simple short URLs
-        if(substr($url,0,3) == 'ftp' && (substr($url,0,6) != 'ftp://')){
+        if (substr($url, 0, 3) == 'ftp' && (substr($url, 0, 6) != 'ftp://')) {
```

Comment:
If `$url` starts with `'ftp'` then of course it doesn't start with `'ftp://'`!  So this could be simplified to:
`if (subset($url, 0, 3) == 'ftp') {` ?

Reply:
Before this step, there is the filtering by https://codesearch.dokuwiki.org/xref/dokuwiki/inc/Parsing/ParserMode/Externallink.php
The only url, which I could found, without `ftp://` which passes and is modified by this was:  `ftp.test.com/` which translated to `ftp://ftp.test.com/`.

Reply:
In my understanding, the second part of the condition should prevent that `ftp://test.com/` becomes `ftp://ftp://test.com/`?

For `www.test.com/`, I agree that it cannot contain the protocol due to the different spelling. So that can be simplified.